ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/ActiveContractKeysPool.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com
+package daml.ledger.api.benchtool.submission
+
+import daml.ledger.api.v1.value.Value
+
+import scala.collection.mutable
+
+/** Keeps track of contract keys of contracts that haven't been used up (archived) yet.
+  * Allows to select the next contract key to use up at random.
+  */
+final class ActiveContractKeysPool(randomnessProvider: RandomnessProvider) {
+
+  private val poolsPerTemplate = mutable.Map.empty[String, DepletingUniformRandomPool[Value]]
+
+  def getAndRemoveContractKey(templateName: String): Value = synchronized {
+    val pool = poolsPerTemplate(templateName)
+    pool.pop()
+  }
+
+  def addContractKey(templateName: String, key: Value): Unit = synchronized {
+    if (!poolsPerTemplate.contains(templateName)) {
+      poolsPerTemplate.put(templateName, new DepletingUniformRandomPool(randomnessProvider))
+    }
+    val pool = poolsPerTemplate(templateName)
+    pool.put(key)
+  }
+}
+
+/** A pool of elements supporting two operations:
+  * 1. pop() - select an element uniformly at random and remove it from the pool.
```

Comment:
it seems that the put is random, not the pop

Reply:
Or you can say that the you get a random element wrt to insertion order. But I would specify that

Reply:
I think the observable behavior matches this description. 
Also, I think it is more natural the describe the randomness in the `pop()` method because this is when the users of this API will experience the randomness. `put()` on the other hand has no observable behavior